### PR TITLE
python packaging

### PR DESCRIPTION
**Pull request by ghost**
This pull request makes makes couple of changes that will be helpful if we ever want to provide this as a python package via pip or something.

1) Use setuptools instead of autoconf, makefile way of compiling library.
2) Make mpow.c a python extension rather than a C library.

### CODE REVIEW

**Comment by ghost** [comment]
3) Python dependencies and several meta info can now be specified in setup.py. Some of the fields need info that I don't have.

**Comment by ghost** [comment]
i just realised something when I was trying to test the setup configuration. setuptools will try to build the extensions first. So for the people who want to use the work server, an extension free build option should be available. But the decision we have to make is, since the library is performing just as fast as the server, should we continue providing that option?

**Comment by guilhermelawless** [comment]
I will setup a quick performance script to check your claims that they are as fast.

**Comment by guilhermelawless** [comment]
It's building successfully with GPU for me, but then it doesn't use it in the client, how come?

**Comment by guilhermelawless** [comment]
@jamescoxon  could you fill the details in `setup.py` ? Thanks! This setup is definitely an improvement.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    macros = [('HAVE_CL_CL_H', '1')]
+    if sys.platform == 'darwin':
+        macros = [('HAVE_OPENCL_OPENCL_H', '1')]
+        ela=['-framework', 'OpenCL']
+else:
+    libs = ['b2']
+    eca = ['-fopenmp']
+
+setup(
+    name="nano-dpow",
```
Comment (guilhermelawless) [inline setup.py]: **nano-dpow-client** perhaps

**Review by jamescoxon — APPROVED** [review]

**Pull request merged**